src/simplnx/DataStructure/BaseGroup.cpp

```diff
@@ -133,19 +135,21 @@ bool BaseGroup::isParentOf(const DataObject* dataObj) const
   return std::find_if(origDataPaths.begin(), origDataPaths.end(), [dataObj](const DataPath& path) { return dataObj->hasParent(path); }) != origDataPaths.end();
 }
 
-bool BaseGroup::insert(const std::weak_ptr<DataObject>& obj)
+Result<bool> BaseGroup::insert(const std::weak_ptr<DataObject>& obj)
 {
   auto ptr = obj.lock();
-  if(!canInsert(ptr.get()))
+  auto result = canInsert(ptr.get());
+  if(result.invalid())
   {
-    return false;
+    return result;
   }
   if(m_DataMap.insert(ptr))
   {
     ptr->addParent(this);
-    return true;
+    return {true};
```

Comment:
```suggestion
    return {{true}};
```
Since we use std::expected (from extraneous nonstd library) I believe we need the second bracket to construct it inside the Result object to pass it to the default constructor. This ensures that the valid() and invalid() Result wrapper methods function correctly. @JDuffeyBQ should probably verify I am correct in this behavior since I'm shaky on how the result is intended to function.

Reply:
I think that issue was with a particular version of gcc. Testing on my Windows machine both ways compile and produce the same result.

Also in this particular case `Result<bool>` is redundant since the result already conveys the success or failure so it should just be `Result<void>`.

Reply:
There is a lot of places that this is going to hit. I'm slowly going down this rabbit hole a bit at a time.

@JDuffeyBQ Ironically I started off with a Result<> but then changed to Result<bool> thinking that is what was needed. But you are correct in the fact that you return a "MakeErrorResult" will set the result to invalid, so that works.